lib/index.js

```diff
@@ -12,10 +12,23 @@ const DataStore = require("./datastore"),
  * for use.  This method calls {@link DataStore#prepare}, returning the
  * (promised) DataStore once it's ready for use.
  *
+ * The signature for `{recordMetric}` (if provided) should conform to
+ *  the following:
+ *
+ * ```
+ * async function recordMetric(method, id, fields) {
+ *   // {method} is one of "added" | "updated" | "deleted"
+ *   // {id} is the item.id
+ *   // {fields} either:
+ *   // * array strings denoting which fields were changed (if method === "updated")
+ *   // * undefined (otherwise)
+ * }
+ * ```
+ *
  * @param {Object} [cfg] - The configuration parameters
  * @param {String} [cfg.bucket="lockbox"] - The bucket to persist data to
- * @param {Object} [cfg.prompts] - The initial set of callback {@link
- *                 DataStore#prompts} for this DataStore.
+ * @param {Function} [cfg.recordMetric] - The function to record item metrics
+ *        events
```

Comment:
I don't have a real preference, but I wonder if we could do this more on the front-end side with code like this in the extension:

```js
await [result, diff] = datastore.add(...);
recordMetric(diff);
```

That might let us keep the datastore simpler. However, I'm fine with the current approach too.